backend/substrapp/tasks/tasks.py

```diff
@@ -640,8 +640,10 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
         chainkeys_directory = get_chainkeys_directory(compute_plan_id)
 
         if not os.path.exists(chainkeys_directory):
+            cp_metadata = get_object_from_ledger(compute_plan_id, 'queryComputePlan')
+            cp_tag = cp_metadata['tag']
             secret_namespace = os.getenv('K8S_SECRET_NAMESPACE', 'default')
-            label_selector = f"compute_plan={subtuple.get('tag')}"
+            label_selector = f"compute_plan={cp_tag}"
```

Comment:
The tag is not associated to the tuple ?
If it's on the compute this part should be moved outside this function which should not do a ledger call

Reply:
The tag is associated with the compute plan (cf PR description).

Why this function should not do a ledger call? 

I can move the ledger call to the upper function but I'm afraid it will make the code harder to read: this call needs to be done only on specific cases: 
- secrets mode enabled
- and, tuple with compute plan
- and, secrets directory not existing

Reply:
The assumption is that all metadata needed in the `do_task` are already provided as it's a compute task, it may have no connection to any services at all.
It's better to fetch all metadata before, where we could potential do ledger call.

You can just pass the `cp_tag` no matter what and let the strategy in the `do_task`, only the ledger part should be done outside